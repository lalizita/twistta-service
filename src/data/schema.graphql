input CreateTweetInput {
  author: ID!
  parent: String
  text: String!
  clientMutationId: String
}

type CreateTweetPayload {
  tweets: [Tweet]
  error: String
  clientMutationId: String
}

type Mutation {
  UserRegister(input: RegisterUsernameInput!): RegisterUsernamePayload
  UserLogin(input: UserLoginInput!): UserLoginPayload
  CreateTweet(input: CreateTweetInput!): CreateTweetPayload
}

"""Query of users"""
type Query {
  users: [User]
  tweets(id: ID!): [Tweet]
}

input RegisterUsernameInput {
  username: String!
  email: String!
  password: String!
  clientMutationId: String
}

type RegisterUsernamePayload {
  username: String
  token: String
  clientMutationId: String
}

"""Tweets of a user"""
type Tweet {
  author: ID!
  parent: ID
  text: String
}

"""user data"""
type User {
  _id: String
  username: String!
  active: Boolean
  email: String!
}

input UserLoginInput {
  username: String!
  password: String!
  clientMutationId: String
}

type UserLoginPayload {
  token: String
  error: String
  clientMutationId: String
}
