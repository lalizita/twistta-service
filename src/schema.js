import { GraphQLSchema } from 'graphql';

import QueryType from './type/QueryType';
import MutationType from './type/MutationType';

console.log(MutationType)

export const schema = new GraphQLSchema({
  query: QueryType,
  mutation: MutationType
});