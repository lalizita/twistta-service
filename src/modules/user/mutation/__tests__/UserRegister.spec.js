import { graphql } from 'graphql';
import { connectMongoose } from '../../../../../test/helper';
import { schema } from '../../../../schema'

describe('User register test', () => {
  beforeAll(async() => {
    await connectMongoose();
  })
  it('Should create a user', async () => {
    const userObject = {
      username: 'lais_registrada',
      email: 'lais@mail.com',
      password: '123234'
    }

    const query =`
      mutation M(
        $username: String!
        $email: String!
        $password: String!
      ) {
        UserRegister(input: {
          username: $username
          email: $email
          password: $password
        }) {
          username
        }
      }
    `
    const variables = userObject
    const { data: { UserRegister } } = await graphql(schema, query, {}, {}, variables)
    expect(UserRegister).toMatchSnapshot();
    expect(UserRegister.username).toBe(userObject.username)
  })
})