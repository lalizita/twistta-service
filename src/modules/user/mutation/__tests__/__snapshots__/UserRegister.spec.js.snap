// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User register test Should create a user 1`] = `
Object {
  "username": "lais_registrada",
}
`;
